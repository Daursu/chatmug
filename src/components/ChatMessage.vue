<template>
  <div class="flex text-left mb-4 pl-2" :class="{ 'flex-row-reverse pr-2': ownMessage }">
    <div class="mt-2">
      <ChatUserGravatar :username="message.username"/>
    </div>
    <div class="flex-grow ml-3 max-w-xs mr-3">
      <div class="shadow border rounded text-sm bg-white p-3"
           :class="{ 'bg-green text-white': ownMessage }">
        <div class="text-left font-bold mb-2">
          {{ message.username }}
        </div>
        {{ message.message }}
      </div>
      <div class="text-right italic text-xs mt-2">
        <span :title="isoDate">{{ date }}</span>
      </div>
    </div>
  </div>
</template>

<script>
import moment from 'moment';
import ChatUserGravatar from '@/components/ChatUserGravatar.vue';

export default {
  name: 'ChatMessage',

  components: {
    ChatUserGravatar,
  },

  props: {
    message: Object,
    ownMessage: Boolean,
  },

  computed: {
    date() {
      return moment(this.message.timestamp).format('hh:mm a');
    },

    isoDate() {
      return moment(this.message.timestamp).toISOString();
    },
  },
};
</script>

<style scoped lang="scss">
  img {
    max-width: 2em;
    max-height: 2em;
  }
</style>
