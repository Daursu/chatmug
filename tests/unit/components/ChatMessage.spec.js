import { shallowMount } from '@vue/test-utils';
import ChatMessage from '@/components/ChatMessage.vue';

describe('ChatMessage.vue', () => {
  const message = {
    id: '2c85d5c0-8c8e-4e38-90bc-0c57aaca70ed',
    username: 'John',
    message: 'Hello ChatMug',
    timestamp: 1549879200000, // 10:00 am UTC
  };

  it('renders the element ok', () => {
    const wrapper = shallowMount(ChatMessage, {
      propsData: {
        message,
        ownMessage: false,
      }
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('renders the component', () => {
    const wrapper = shallowMount(ChatMessage, {
      propsData: {
        message,
        ownMessage: false,
      }
    });

    expect(wrapper.html()).toContain('10:00 am');
    expect(wrapper.html()).toContain('John');
    expect(wrapper.html()).toContain('Hello ChatMug');
  });

  it('uses a green background for own messages', () => {
    const wrapper = shallowMount(ChatMessage, {
      propsData: {
        message,
        ownMessage: true,
      }
    });

    expect(wrapper.contains('.bg-green')).toBe(true);
  });
});
