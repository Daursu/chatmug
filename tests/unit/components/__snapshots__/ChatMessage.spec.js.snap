// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage.vue renders the element ok 1`] = `
<div class="flex text-left mb-4 pl-2">
  <div class="mt-2">
    <chatusergravatar-stub username="John"></chatusergravatar-stub>
  </div>
  <div class="flex-grow ml-3 max-w-xs mr-3">
    <div class="shadow border rounded text-sm bg-white p-3">
      <div class="text-left font-bold mb-2">
        John
      </div>
      Hello ChatMug
    </div>
    <div class="text-right italic text-xs mt-2"><span title="2019-02-11T10:00:00.000Z">10:00 am</span></div>
  </div>
</div>
`;
